packages/ui/src/forum/components/PostList/PostListItem.tsx

```diff
@@ -66,8 +67,9 @@ export const PostListItem = ({
   }, [ref.current])
 
   useEffect(() => {
-    if (ref.current && isSelected) {
+    if (ref.current && isSelected && firstLoad) {
       ref.current.scrollIntoView({ behavior: 'smooth', inline: 'start' })
+      setFirstLoad(false)
     }
   }, [location, ref.current])
```

Comment:
This solution would work but it add some unnecessary complexity.

Instead AFAICS both `location` and `ref.current` are the wrong dependencies `useEffect`.
1. `location` is actually the source of the problem because the `useLocation` hook returns a new JS object (same values different reference) on every rerender.
     Instead it should be:
    ```ts
    return useMemo(() => ({ pathname, search, hash, origin }), [pathname, search, hash])
    ```
     But in this case we should only care about `isSelected` (which itself depends on the location).

3. `ref.current` is assigned to the `ForumPostBlock` `div` before any of the `useEffect` get called, so it's useless to add it in the dependencies.
```suggestion
    if (ref.current && isSelected) {
      ref.current.scrollIntoView({ behavior: 'smooth', inline: 'start' })
    }
  }, [isSelected])
```
**Improtant:**
Please fix the `useLocation` hook (as described above) because it might cause unnecessary re-renders in other places and maybe some other bugs.

Reply:
You right. Resolved

Reply:
Done. Sorry misunderstood this part.

Reply:
Np!